master/pyext/src/validation/__init__.py

```diff
@@ -12,527 +12,623 @@
-            mol_name=strc
+            mol_name = strc
         return mol_name
-    
+
     def check_sphere(self):
-        """check resolution of structure,returns 0 if its atomic and 1 if the model is multires"""
-        spheres=[len(b._spheres) for i in self.system.state_groups for j in i for a in j for b in a]
+        """check resolution of structure,
+        returns 0 if its atomic and 1 if the model is multires"""
+        spheres = [len(b._spheres) for i in self.system.state_groups
+                   for j in i for a in j for b in a]
         if 0 not in spheres:
             return 1
         else:
-            return 0 
+            return 0
 
-    def get_assembly_ID_of_models(self)->list:
+    def get_assembly_ID_of_models(self) -> list:
         """Assembly info i.e. model assemblies in the file """
-        assembly_id=[b.assembly._id for i in self.system.state_groups for j in i for a in j for b in a]
+        assembly_id = [
+            b.assembly._id for i in self.system.state_groups
+            for j in i for a in j for b in a]
         return assembly_id
 
-    def get_representation_ID_of_models(self)->list:
+    def get_representation_ID_of_models(self) -> list:
         """Number of representations in model """
-        representation_id=[b.representation._id for i in self.system.state_groups for j in i for a in j for b in a]
+        representation_id = [
+            b.representation._id for i in self.system.state_groups
+            for j in i for a in j for b in a]
         return representation_id
 
-    def get_model_names(self)->list:
+    def get_model_names(self) -> list:
         """ Names of models"""
-        model_name1=[a.name for i in self.system.state_groups for j in i for a in j]
-        model_name2=[b.name for i in self.system.state_groups for j in i for a in j for b in a]
-        if len(model_name1)==len(model_name2):
-            model_name =[str(t[0])+'/'+str(t[1]) for t in zip(model_name1,model_name2)]
+        model_name1 = [
+            a.name for i in self.system.state_groups for j in i for a in j]
+        model_name2 = [
+            b.name for i in self.system.state_groups
+            for j in i for a in j for b in a]
+        if len(model_name1) == len(model_name2):
+            model_name = [str(t[0])+'/'+str(t[1])
+                          for t in zip(model_name1, model_name2)]
         else:
-            model_name=model_name2
+            model_name = model_name2
         return model_name
-    
-    def get_model_assem_dict(self)->dict:
+
+    def get_model_assem_dict(self) -> dict:
         """Map models to assemblies """
-        model_id=map(int,[b._id for i in self.system.state_groups for j in i for a in j for b in a])
-        assembly_id=map(int,self.get_assembly_ID_of_models())
-        model_assembly=dict(zip(model_id,assembly_id))
+        model_id = map(int, [b._id for i in self.system.state_groups
+                             for j in i for a in j for b in a])
```

Comment:
It's *usually* cleaner to use a list comprehension than `map`, e.g. 

    model_id = [int(b._id) for i in self.system.state_groups
                         for j in i for a in j for b in a])